src/compas_fab/robots/configuration.py

```diff
@@ -62,28 +110,132 @@ class Configuration(object):
 
     """
 
-    def __init__(self, values=None, types=None, joint_names=None):
+    def __init__(self, joint_values=None, joint_types=None, joint_names=None):
+        joint_values = FixedLengthList(joint_values or [])
+        joint_types = FixedLengthList(joint_types or [])
+        joint_names = FixedLengthList(joint_names or [], validator=joint_names_validator)
+
+        if len(joint_values) != len(joint_types):
+            raise ValueError('{} joint_values must have {} joint_types, but {} given.'.format(
+                len(joint_values), len(joint_values), len(joint_types)))
+
+        if joint_names and len(joint_names) != len(joint_values):
+            raise ValueError('{} joint_values must have either 0 or {} names, but {} given'.format(
+                len(joint_values), len(joint_values), len(joint_names)
+            ))
+
         self._precision = '3f'
```

Comment:
Should this actually be `compas.PRECISION`?

Reply:
Is it used for anything other than the `__str_`? If that's the only usage, then, no need to.